eventq_aws/lib/eventq_aws/aws_queue_worker.rb

```diff
@@ -77,7 +77,8 @@ def start_process(options, queue, block)
         @thread_count.times do
           thr = Thread.new do
 
-            client = options[:client]
+            client = options[:client] || new_client_instance(options) # singleton or non-singleton
```

Comment:
do we really need to be backwards compatible?  This just allows a client that might not be thread safe to be introduced.
I think we should not even have this option.